go/private/context.bzl

```diff
@@ -59,20 +60,22 @@ _LINKER_OPTIONS_BLACKLIST = {
 def _filter_options(options, blacklist):
   return [option for option in options if option not in blacklist]
 
-def _child_name(go, path, ext, name):
-  childname = mode_string(go.mode) + "/"
-  childname += name if name else go._ctx.label.name
-  if path:
-    childname += "~/" + path
-  if ext:
-    childname += ext
-  return childname
+def _child_name(go, path, ext, name, output_path):
```

Comment:
See comments in `emit_binary`.

These functions should be left alone to generate mode-specific paths without user customization.

Non-mode-specific paths can still be declared with `ctx.actions.declare_file`.